mypy/modulefinder.py

```diff
@@ -0,0 +1,397 @@
+"""Low-level infrastructure to find modules.
+
+This build on fscache.py; find_sources.py builds on top of this.
+"""
+
+import ast
+import collections
+import functools
+import os
+import subprocess
+import sys
+
+from typing import Dict, List, NamedTuple, Optional, Set, Tuple
+
+MYPY = False
+if MYPY:
+    from typing_extensions import Final
+
+from mypy.defaults import PYTHON3_VERSION_MIN
+from mypy.fscache import FileSystemCache
+from mypy.options import Options
+from mypy import sitepkgs
+
+# Paths to be searched in find_module().
+SearchPaths = NamedTuple(
+    'SearchPaths',
+    [('python_path', Tuple[str, ...]),  # where user code is found
+     ('mypy_path', Tuple[str, ...]),  # from $MYPYPATH or config variable
+     ('package_path', Tuple[str, ...]),  # from get_site_packages_dirs()
+     ('typeshed_path', Tuple[str, ...]),  # paths in typeshed
+     ])
+
+# Package dirs are a two-tuple of path to search and whether to verify the module
+OnePackageDir = Tuple[str, bool]
+PackageDirs = List[OnePackageDir]
+
+PYTHON_EXTENSIONS = ['.pyi', '.py']  # type: Final
+
+
+class BuildSource:
+    """A single source file."""
+
+    def __init__(self, path: Optional[str], module: Optional[str],
+                 text: Optional[str], base_dir: Optional[str] = None) -> None:
+        self.path = path  # File where it's found (e.g. 'xxx/yyy/foo/bar.py')
+        self.module = module or '__main__'  # Module name (e.g. 'foo.bar')
+        self.text = text  # Source code, if initially supplied, else None
+        self.base_dir = base_dir    # Directory where the package is rooted (e.g. 'xxx/yyy')
```

Comment:
Remove extra spaces before this comment?

Reply:
Will do.